azure_functions_worker/dispatcher.py

```diff
@@ -313,82 +322,102 @@ async def _handle__worker_status_request(self, request):
-                    function_metadata_response=protos.FunctionMetadataResponse(
-                        use_default_metadata_indexing=True,
-                        result=protos.StatusResult(
-                            status=protos.StatusResult.Success)))
+            self._function_metadata_result = (
+                self.index_functions(function_path)) \
+                if os.path.exists(function_path) else None
 
-            fx_metadata_results = self.index_functions(function_path)
+        except Exception as ex:
+            self._function_metadata_exception = ex
 
+    async def _handle__functions_metadata_request(self, request):
+        metadata_request = request.functions_metadata_request
+        function_app_directory = metadata_request.function_app_directory
+
+        if not is_envvar_true(PYTHON_ENABLE_INIT_INDEXING):
+            self.load_function_metadata(
+                function_app_directory,
+                caller_info="functions_metadata_request")
+
+        if self._function_metadata_exception:
             return protos.StreamingMessage(
-                request_id=request.request_id,
+                request_id=self.request_id,
                 function_metadata_response=protos.FunctionMetadataResponse(
-                    function_metadata_results=fx_metadata_results,
                     result=protos.StatusResult(
-                        status=protos.StatusResult.Success)))
+                        status=protos.StatusResult.Failure,
+                        exception=self._serialize_exception(
+                            self._function_metadata_exception))))
+        else:
+            metadata_result = self._function_metadata_result
 
-        except Exception as ex:
             return protos.StreamingMessage(
-                request_id=self.request_id,
+                request_id=request.request_id,
                 function_metadata_response=protos.FunctionMetadataResponse(
+                    use_default_metadata_indexing=False if metadata_result else
+                    True,
+                    function_metadata_results=metadata_result,
                     result=protos.StatusResult(
-                        status=protos.StatusResult.Failure,
-                        exception=self._serialize_exception(ex))))
+                        status=protos.StatusResult.Success)))
 
     async def _handle__function_load_request(self, request):
         func_request = request.function_load_request
         function_id = func_request.function_id
         function_metadata = func_request.metadata
         function_name = function_metadata.name
+        function_app_directory = function_metadata.directory
 
         logger.info(
             'Received WorkerLoadRequest, request ID %s, function_id: %s,'
-            'function_name: %s,', self.request_id, function_id, function_name)
+            'function_name: %s,',
+            self.request_id, function_id, function_name)
```

Comment:
Would it help to add the information that it was already found in the `self._functions.get_function(function_id)` call in the log itself? To see the scenario on how much this is useful?

Reply:
Fixed.. Added variable function_already_indexed